lib/linters.py

```diff
@@ -0,0 +1,627 @@
+    pass
+
+  def check_service(self, service):
+    """May be implemented to check a given service found in a .svc file"""
+    pass
+
+  def check_networks(self, networks):
+    """May be implemented to check all of the networks found in a .net file"""
+    pass
+
+  def check_services(self, services):
+    """May be implemented to check all of the services found in a .svc file"""
+    pass
+
+  def check_term(self, policy, header, term):
+    """May be implemented to check the given term under a given header"""
+    pass
+
+
+@register_linter
+class AddressNumEnforce(BaseLintRule):
+  """Warns about terms with more than a few address tokens."""
+
+  def check_term(self, policy, header, term):
+    addr_attrs = (
+      'source_address',
+      'destination_address',
+      'destination_address_exclude',
+      'source_address_exclude',
+    )
+    for attrname in addr_attrs:
+      addrs = getattr(term, attrname)
+      parents = set([addr.parent_token for addr in addrs])
+      if len(parents) > 4:
+        msg = ("term %s has too many objects in" % term.name +
+             " one of the address fields. " +
+             "Please review and consolidate the addresses")
+        self.add(Severity.WARNING, msg, term.name.lineno)
+
+
+@register_linter
+class RegexNameEnforcer(BaseLintRule):
+  """Ensures objects and terms following naming conventions."""
+
+  DEFAULTS = {
+    'NETNAME': r'^([A-Z][A-Z0-9_]+|h_[0-9a-f\.]+|n_[0-9a-f\.]+_[0-9]+)$',
+    'SVCNAME': r'^((TCP|UDP|TCP_UDP)_[0-9]+(-[0-9]+)?|[A-Z][A-Z0-9_]+)$',
+    'TERMNAME': {
+      'default': r'^[a-z][a-z0-9\-\.]+$',
+      'srx': r'^[a-z][a-z0-9\-]+$',
+    }
+  }
+
+  def setup(self):
+    self.netname_re = re.compile(self.config['NETNAME'])
+    self.svcname_re = re.compile(self.config['SVCNAME'])
+    self.termname_re = {k: re.compile(v) for k, v
+              in self.DEFAULTS['TERMNAME'].items()}
+
+  def check_network(self, network):
+    if not self.netname_re.match(network.name):
+      msg = '%s is not a valid network name' % network.name
+      self.add(Severity.WARNING, msg, network.name.lineno)
+
+  def check_service(self, service):
+    if not self.svcname_re.match(service.name):
+      msg = '%s is not a valid service name' % service.name
+      self.add(Severity.WARNING, msg, service.name.lineno)
+
+  def check_term(self, policy, header, term):
+    header_targets = {}
+    if header:
+      header_targets = {x.platform for x in header.target}
+    for target, re_pattern in self.termname_re.iteritems():
+      if target == 'default' and not re_pattern.match(term.name):
+        msg = ('%s is not a valid '
+             'term name for all platforms' % term.name)
+        self.add(Severity.ERROR, msg, term.name.lineno)
+      elif target in header_targets and not re_pattern.match(term.name):
+        msg = '%s is not a valid term name for target: %s' % (term.name,
+                                    target)
```

Comment:
Please fix this indentation also. How it is now you would need to add 26 spaces but you could follow the example above to do it differently.